fix: properly use logging configuration in Executor, Gateway and Client (#5638)
jina-ai/jina: docs/concepts/flow/basics.md

## Start and stop

> When a {class}`~jina.Flow` starts, all its {ref}`added Executors <flow-add-executors>` will start as well, making it
> possible to {ref}`reach the service through its API <third-party-client>`.

Reviewer: ```suggestion
When a {class}`~jina.Flow` starts, all its {ref}`added Executors <flow-add-executors>` will start as well, making it possible to {ref}`reach the service through its API <third-party-client>`.
```

```
````

> The statement `with f:` starts the Flow, and exiting the indented `with` block stops the Flow, including all Executors
> defined in it.

Reviewer: ```suggestion
The statement `with f:` starts the Flow, and exiting the indented `with` block stops the Flow, including all Executors defined in it.
```

A successful start of a Flow looks like this:

```{figure} success-flow.png
:scale: 70%
```

> Your addresses and entrypoints can be found in the output. When you enable more features such as monitoring, HTTP
> gateway, TLS encryption, this display expands to contain more information.

Reviewer: ```suggestion
Your addresses and entrypoints can be found in the output. When you enable more features such as monitoring, HTTP gateway, TLS encryption, this display expands to contain more information.
```
Lot of oddly placed linebreaks throughout. Can you fix?

### Serve on Google Colab

> Google Colab provides an easy-to-use Jupyter notebook environment with GPU/TPU support. Flow is fully compatible with

Reviewer: Add a link to Colab

```yaml
handlers:
  - StreamHandler
level: INFO
configs:
  StreamHandler:
    format: '%(asctime)s:{name:>15}@%(process)2d[%(levelname).1s]:%(message)s'
    formatter: JsonFormatter
```

The logging configuration can be used as follows:

````{tab} Python
```python
from jina import Flow

f = Flow(log_config='./logging.json.yml')
```
````

````{tab} YAML
```yaml
jtype: Flow
with:
    log_config: './logging.json.yml'
```
````

### Overriding logging configuration

The default logging or custom logging configuration at the `Flow` level will be propagated to the `Gateway` and `Executor` entities.
If that is not desired, every `Gateway` or `Executor` entity can be provided with a custom logging configuration. 

You can configure two different `Executors` as in the below example:

```python
from jina import Flow

f = (
    Flow().add(log_config='./logging.json.yml').add(log_config='./logging.file.yml')
)  # Create a Flow with two Executors
```

`logging.file.yml` is another YAML file with a custom `FileHandler` configuration.

````{hint}
Refer to {ref}`Gateway logging configuration <gateway-logging-configuration>` section for configuring the `Gateway` logging.
````

````{caution}

> When exporting the Flow to Kubernetes, the log_config file path must refer to the absolute local path of each container. The custom logging

Reviewer 2: also applicable to dockerized executors or jcloud

Reviewer 3: When `Dockerized Executors` exists u could pass them via `volumes` from your `host`

Author: Shall I add another sentence to the end like below?
```
When running the Flow on local machine with dockerized `Executor`, the custom logging file can be mounted using the `volumes` argument. 
```

Reviewer 3: maybe, this would be more correct than saying it is not supported

{ref}`Kubernetes <kubernetes>`.
```

(logging-configuration)=

## Logging Configuration

> The default {class}`jina.logging.logger.JinaLogger` uses rich console logging set up that writes to the system console.
> The `log_config` argument can be used to pass in a string of the pre-configured logging configuration names in Jina or
> the absolute YAML file path of the custom logging configuration. For most cases, the default logging configuration
> sufficiently covers local, Docker and Kubernetes environments.

Reviewer: ```suggestion
The default {class}`jina.logging.logger.JinaLogger` uses rich console logging that writes to the system console.
The `log_config` argument can be used to pass in a string of the pre-configured logging configuration names in Jina or
the absolute YAML file path of the custom logging configuration. For most cases, the default logging configuration
sufficiently covers local, Docker and Kubernetes environments.
```

Custom logging handlers can be configured by following
the Python official [Logging Cookbook](https://docs.python.org/3/howto/logging-cookbook.html#logging-cookbook) examples.
An example custom logging configuration file defined in a YAML file `logging.json.yml` is:

```yaml
handlers:
  - StreamHandler
level: INFO
configs:
  StreamHandler:
    format: '%(asctime)s:{name:>15}@%(process)2d[%(levelname).1s]:%(message)s'
    formatter: JsonFormatter
```

The logging configuration can be used as follows:

````{tab} Python
```python
from jina import Flow

f = Flow(log_config='./logging.json.yml')
```
````

````{tab} YAML
```yaml
jtype: Flow
with:
    log_config: './logging.json.yml'
```
````

### Overriding logging configuration

The default logging or custom logging configuration at the `Flow` level will be propagated to the `Gateway` and `Executor` entities.

> If that is not desired, every `Gateway` or `Executor` entity can be provided a custom logging configuration. 

Reviewer: ```suggestion
If that is not desired, every `Gateway` or `Executor` entity can be provided with a custom logging configuration. 
```

> Two different `Executors` can be configured as in the below example.

Reviewer: ```suggestion
You can configure two different `Executors` as in the below example:
```

```python
from jina import Flow

f = (
    Flow().add(log_config='./logging.json.yml').add(log_config='./logging.file.yml')
)  # Create a Flow with two Executors
```

> The `logging.file.yml` is another YAML file with a custom `FileHandler` configuration.

Reviewer: ```suggestion
`logging.file.yml` is another YAML file with a custom `FileHandler` configuration.
```
